### ISSUE TITLE
Using js_of_ocaml more directly from OCaml code itself, do read. 

### ISSUE THREAD

**Issue by fxfactorial** [ocsigen/js_of_ocaml#498]
I want to use js_of_ocaml directly as a compiler but with all the hassles of findlib handled for me and at the source level, thereby skipping the compiling to byte code hassles/abuses of build systems. I think this would be a killer feature and it will let me do amazing things. Here is a signature that I'm looking for: 

```ocaml
val javascript_from_ocaml : 
(* Let me do (object%js val count = 10 end) and count becomes a top level var count = 10; *) 
?js_global_values: 'a Js.t -> 
(* of course we want to use js_of_ocaml.ppx by default *)
?(ocaml_pkgs=["js_of_ocaml.ppx"]) ->
(* self-explanatory *)
?(minified=true)
(* Only include JavaScript, no linking of the Dom API at all, only what's limited to the JavaScript language *)
?(just_javascript=true)
(* These are OCaml .ml files *)
src_files:string list  -> 
unit ->
(* Resultant JavaScript as one long string *)
string
```

I realize this could be hacked together in various ways but I think `js_of_ocaml` ought to provide this

Basically here's my idea: 

https://gist.github.com/fxfactorial/8f59fd3e45cc82231263a58bb4e71af1

**Comment by hhugo** [comment]
I don't understand the interaction between `js_global_values: 'a Js.t` and the resulting string.

Also, what about dependencies in ml files ?

Do I understand correctly that you want `findlib` + `the ocaml compiler` ( + `ocamldep` ?) + `the js_of_ocaml compiler` package together ?

**Comment by fxfactorial** [comment]
yes, I would want js_of_ocaml to handle all the dependencies as well.

**Comment by hhugo** [comment]
what about `js_global_values` ? 
I seems to be a lot of work for a _very_ specific use case .. feel free to propose a PR.
I wonder if one can extend https://github.com/janestreet/ocaml_plugin for this

**Comment by fxfactorial** [comment]
it will open up a lot of possibility
